pkg/controller/remotecluster/controller.go

```diff
@@ -265,11 +266,15 @@ func doReconcile(
 		results.WithResults(reconcileAPIKeys(ctx, r.Client, activeAPIKeys, remoteServer, remoteClient, remoteClusterRefs, esClient, r.keystoreProvider))
 	}
 
-	if remoteServerSupportsClusterAPIKeys.IsTrue() {
+	if remoteServerSupportsClusterAPIKeys.IsTrue() { //nolint:nestif
 		// **************************************************************
 		// Delete orphaned API keys from clusters which have been deleted
 		// **************************************************************
 		for _, activeAPIKey := range activeAPIKeys.APIKeys {
+			// Skip API keys managed by the autoops controller.
+			if autoops.IsManagedByAutoOps(activeAPIKey.Metadata) {
```

Comment:
Maybe mention that we are also using a different naming convention so this is relaly just an additional safeguard.